certbot-nginx/tests/parser_test.py

```diff
@@ -482,7 +482,45 @@ def test_duplicate_vhost_remove_ipv6only(self):
                 called = True
         self.assertTrue(called)
 
+    def test_valid_unicode_characters(self):
+        nparser = parser.NginxParser(self.config_path)
+        path = nparser.abs_path('valid_unicode_comments.conf')
+        parsed = nparser._parse_files(path)  # pylint: disable=protected-access
+        self.assertEqual(['server'], parsed[0][2][0])
+        self.assertEqual(['listen', '80'], parsed[0][2][1][3])
+
+    def test_invalid_unicode_characters(self):
+        with self.assertLogs() as log:
+            nparser = parser.NginxParser(self.config_path)
+            path = nparser.abs_path('invalid_unicode_comments.conf')
+            parsed = nparser._parse_files(path)  # pylint: disable=protected-access
+
+        self.assertEqual([], parsed)
+        self.assertTrue([
```

Comment:
I don't think this use of `assertTrue` is valid:

First, the thing you're asserting is `True` is a list of booleans, which is not the right type to use here.

Second, if there are multiple items in `log.output` (I don't remember whether that can happen), you will get a `True` for each one that is right but no list entry for each one that is wrong.

I would say that you could do something like

`assertTrue(all(('invalid character' in output) and ('UTF-8' in output) for output in log.output))`

if you expect multiple items in `log.output` (or `any()` instead of `all()` if only one of them needs to have the intended output).

Reply:
Oh this is definitely my mistake. I should've apply `any()` as you suggested. Fixed in a48907920885671ef5a57bc1cda556e90c5e0e20.